src/main/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsv.java

```diff
@@ -155,26 +156,48 @@ private static void exportFieldValues(final PDFInteractiveForm form, final CSVPr
         while (fieldIterator.hasNext()) {
             final PDFField field = fieldIterator.next();
 
-            final String formattedValue = field.getFormattedValue();
-            if (formattedValue == null) {
-                final List value = field.getValueList();
-                if (value == null) {
-                    printer.print("");
-                    if (LOGGER.isLoggable(Level.WARNING)) {
-                        LOGGER.warning(field.getQualifiedName() + " has no value!");
-                    }
+            List value;
+            if (VersionUtils.pdfjtIsBeforeVersion4()) {
+                value = field.getValueList();
+                validateField(printer, field, value);
+            } else {
+                final String formattedValue = field.getFormattedValue();
```

Comment:
@datalogics-bhaugen I could create a method in VersionUtils that checks specifically for version 4.7.0 . Is that what you were thinking?